src/Auth.spec.ts

```diff
@@ -1381,6 +1383,66 @@ describe('Auth', () => {
     assert.strictEqual((requestStub.lastCall.args[0] as any).headers['x-anonymous'], true);
   });
 
+  it('calls api with correct params using federated identity flow from GitHub Action', async () => {
+    process.env = {
+      ACTIONS_ID_TOKEN_REQUEST_URL: 'https://pipelinesghubeus2.actions.githubusercontent.com/P0Hx2m9SFWNP5AeJ8UVAwq97ADEUmaXrYvynThDBYkADVfIDh4/00000000-0000-0000-0000-000000000000/_apis/distributedtask/hubs/Actions/plans/485ea249-6e9d-4c8d-8853-c57725625712/jobs/97ef1aa6-b90f-40f0-81a4-9120b120ceaa/idtoken?api-version=2.0',
+      ACTIONS_ID_TOKEN_REQUEST_TOKEN: 'eyJ0eXAiOiJKV1QiLCJ-GitHubFederationToken-...'
+    };
+    const federatedIdentityClientAssertion = 'eyJ0eXAiOiJKV1QiLCJ-GitHubFederationClientAssertion-...';
+    const accessToken = 'eyJ0eXAiOiJKV1QiLCJ-EntraIDAccessToken...';
+
+    const requestGetStub = sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${process.env.ACTIONS_ID_TOKEN_REQUEST_URL}&audience=${encodeURIComponent(federatedIdentityAudience)}`) {
+        return {
+          "value": federatedIdentityClientAssertion
+        };
+      }
+      throw { error: { "error": "Invalid GET Request" } };
+    });
+
+    const requestPostStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://login.microsoftonline.com/${tenant}/oauth2/v2.0/token`) {
+        return {
+          "access_token": accessToken,
+          "client_id": "a04566df-9a65-4e90-ae3d-574572a16423",
+          "expires_in": "86399",
+          "expires_on": "1587847593",
+          "ext_expires_in": "86399",
+          "not_before": "1587760893",
+          "resource": "https://contoso.sharepoint.com/",
+          "token_type": "Bearer"
+        };
+      }
+
+      throw { error: { "error": "Invalid POST Request" } };
+    });
+
+    auth.connection.authType = AuthType.FederatedIdentity;
+    auth.connection.appId = appId;
+    auth.connection.tenant = tenant;
+
+    const ensuredAccessToken = await auth.ensureAccessToken(resource, logger, true);
+    assert.strictEqual(ensuredAccessToken, accessToken);
+    assert.strictEqual(requestGetStub.firstCall.args[0].url, `${process.env.ACTIONS_ID_TOKEN_REQUEST_URL}&audience=${encodeURIComponent(federatedIdentityAudience)}`);
```

Comment:
Since we do a check in the stub on the URL already, we can simplify this to `assert(requestGetStub.calledOnce);`.